src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -82,6 +87,23 @@ static CurlHandler()
             // Feature detection
             s_supportsSSL = (CurlFeatures.CURL_VERSION_SSL & s_curlVersionInfoData.features) != 0;
             s_supportsAutomaticDecompression = (CurlFeatures.CURL_VERSION_LIBZ & s_curlVersionInfoData.features) != 0;
+
+            // We can send any pointer, even null seems to work. But sending a non-null value, just in case
+            if (CurlSslBackend.AllowsSslCallback(s_curlVersionInfoData.ssl_version))
+            {
+                s_clientCertificatePolicy = ClientCertificatePolicy.Callback;
+                VerboseTrace("openSSL compatible");
+            }
+            else if (CurlSslBackend.AllowsClientCertCommonName(s_curlVersionInfoData.ssl_version))
```

Comment:
You are suggesting that we throw when:
1.  user sets client-cert-options to automatic
2.  ssl-backend is not openssl

Did I understand your suggestion correctly?